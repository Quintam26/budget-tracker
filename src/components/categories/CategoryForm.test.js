import CategoryForm from './CategoryForm';
import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

describe('Category form', () => {

  it('Renders add if no category prop', () => {
    const handleComplete = jest.fn();
    const promise = Promise.resolve();
    handleComplete.mockReturnValueOnce(promise);

    const wrapper = mount(<CategoryForm onComplete={handleComplete}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();

    const category = {
      name: 'Car',
      budget: 5000,
      timestamp: null
    };

    wrapper.find('input[name="name"]').simulate('change', {
      target: {
        name: 'name',
        value: category.name
      }
    });

    wrapper.find('input[name="budget"]').simulate('change', {
      target: {
        name: 'budget',
        value: category.budget
      }
    });

    wrapper.find('button').simulate('submit');

    const calls = handleComplete.mock.calls;
    expect(calls.length).toBe(1);
    expect(calls[0][0]).toEqual(category);

    expect(toJSON(wrapper)).toMatchSnapshot();

    return promise  .then(() => {
      wrapper.update();
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

  });

  it('Renders edit if there is category prop', () => {
    const handleComplete = jest.fn();
    const promise = Promise.resolve();
    handleComplete.mockReturnValueOnce(promise);
    const handleCancel = jest.fn();

    const category = { key: 'abc', name: 'car', budget: 5000 };

    const wrapper = mount(<CategoryForm onComplete={handleComplete} onCancel={handleCancel} category={category}/>);

    expect(toJSON(wrapper)).toMatchSnapshot();

  });

});