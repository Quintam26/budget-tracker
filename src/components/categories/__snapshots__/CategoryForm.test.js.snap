// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category form Renders add if no category prop 1`] = `
<CategoryForm
  onComplete={[MockFunction]}
>
  <form
    onSubmit={[Function]}
  >
    <InputControl
      name="name"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          name
          :
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="budget"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          budget
          :
          <input
            name="budget"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Add
      </button>
    </p>
  </form>
</CategoryForm>
`;

exports[`Category form Renders add if no category prop 2`] = `
<CategoryForm
  onComplete={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "budget": 5000,
            "name": "Car",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <InputControl
      name="name"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          name
          :
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="budget"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          budget
          :
          <input
            name="budget"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Add
      </button>
    </p>
  </form>
</CategoryForm>
`;

exports[`Category form Renders add if no category prop 3`] = `
<CategoryForm
  onComplete={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "budget": 5000,
            "name": "Car",
          },
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": Promise {},
        },
      ],
    }
  }
>
  <form
    onSubmit={[Function]}
  >
    <InputControl
      name="name"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          name
          :
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="budget"
      onChange={[Function]}
      value=""
    >
      <p>
        <label>
          budget
          :
          <input
            name="budget"
            onChange={[Function]}
            required={true}
            value=""
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        type="submit"
      >
        Add
      </button>
    </p>
  </form>
</CategoryForm>
`;

exports[`Category form Renders edit if there is category prop 1`] = `
<CategoryForm
  category={
    Object {
      "budget": 5000,
      "key": "abc",
      "name": "car",
    }
  }
  onCancel={[MockFunction]}
  onComplete={[MockFunction]}
>
  <form
    className="category-form"
    onSubmit={[Function]}
  >
    <InputControl
      name="name"
      onChange={[Function]}
      value="car"
    >
      <p
        className="category-p"
      >
        <label
          className="category-l"
        >
          name
          :
          <input
            name="name"
            onChange={[Function]}
            required={true}
            value="car"
          />
        </label>
      </p>
    </InputControl>
    <InputControl
      name="budget"
      onChange={[Function]}
      value={5000}
    >
      <p
        className="category-p"
      >
        <label
          className="category-l"
        >
          budget
          :
          <input
            name="budget"
            onChange={[Function]}
            required={true}
            value={5000}
          />
        </label>
      </p>
    </InputControl>
    <p>
      <button
        className="add-update-button"
        type="submit"
      >
        Update
      </button>
      <button
        className="cancel-button"
        onClick={[MockFunction]}
        type="button"
      >
        Cancel
      </button>
    </p>
  </form>
</CategoryForm>
`;
